### DISCUSSION TITLE
Endpoint with image file on browser

### DISCUSSION THREAD

**Discussion by dro-sh** [mswjs/msw#1502] [Q&A] [answered]
I use msw for browser, latest version. 
I want to mock endpoint which expect form data with file. I log file on client (ui) and on mock. File has a same name and type, but mismatch size and lastModified. Also I read file via FileReader to base64 on client and mock, strings are mismatched:

```ts
type Picture = {
  file: File;
};

rest.put<NewPicture, PathParams<string>, Picture>(
  "/picture",
  async (req, res, ctx) => {
    const { file } = req.body;
    console.log(file); // mismatch from client
    ....
  }
)
```

There is way to mock uploading images?

**Comment by kettanaito** [comment, marked as answer]
Hey, @dro-sh. 

I believe there are some issues with how MSW handles form data at the moment. I'm addressing most of the known issues in #1436 that's going to be released sometime next year. I highly encourage you to migrate to the Beta version and give it a try today. See #1464 for more details. That version has the correct `FormData` handling, which should allow you to mock form uploads as an example. 

Thanks for understanding.

**Reply by dro-sh** [reply]
Thank you for answer! Yes, it helps: I can upload file and get on mock endpoint.
